### DISCUSSION TITLE
Error when reloading/refreshing page

### DISCUSSION THREAD

**Discussion by madrimovganijon** [KevinVandy/material-react-table#1056] [Q&A]
Hello, can you please point me to the line that is causing this and how to resolve it?
It works on the initial load, then if you re-load the page, it throws an error:
Uncaught TypeError: Cannot read properties of undefined (reading 'length')
Thanks.

`
                import { MaterialReactTable,  useMaterialReactTable, } from "material-react-table";
                import { useQuery } from "@tanstack/react-query";
                import axios from "axios";
                import { columns } from "./columns";

                export const Task = () => {
                    const { isLoading, error, data } = useQuery({
                        queryKey: ["task"],
                        queryFn: () =>
                            axios.get("http://localhost:3001/api/task").then((res) => res.data),
                    });

                    const table = useMaterialReactTable({
                        columns,
                        data
                    });

                    if (isLoading) return "Loading...";

                    if (error) return "An error has occurred: " + error.message;

                    return (
                        <>
                            <MaterialReactTable table={table} />{" "}
                        </>
                    );
                };

`

**Comment by KevinVandy** [comment]
Probably because your data is undefined. It needs to be an empty array while loading.

**Reply by madrimovganijon** [reply]
Thanks Kevin, how can I specify that?

**Reply by KevinVandy** [reply]
Follow the examples

**Reply by madrimovganijon** [reply]
ReactQuery example is not working on Sandbox, StackBlitz and CodeSandbox

**Comment by madrimovganijon** [comment]
If you pass options like this, no errors:

            <MaterialReactTable data={data} columns={columns} />

**Comment by madrimovganijon** [comment]
Resolved it by adding initialData = [] to useQuery
`
    const { isPending, error, data } = useQuery({
        queryKey: ["reportData"],
        queryFn: () =>
            Axios.get("http://localhost:3001/api/task").then((res) => res.data),
        initialData: [],
    });
`
